util/encoding/encoding.go

```diff
@@ -594,3 +595,30 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 	}
 	return b, false
 }
+
+// EncodeTime encodes a time value, appends it to the supplied buffer,
+// and returns the final buffer. The encoding is guaranteed to be ordered
+// Such that if t1.Before(t2) then after EncodeTime(b1, t1), and
+// EncodeTime(b2, t1), Compare(b1, b2) < 0. The time zone offset not
+// included in the encoding.
+func EncodeTime(b []byte, t time.Time) []byte {
+	// Read the unix absolute time. This is the absolute time and is
+	// not time zone offset dependent.
+	sec := t.Unix()
+	nsec := t.Nanosecond()
+	if nsec < 0 {
+		// if it were negative we would need to change the way we encode.
```

Comment:
I was a bit confused in my head.

when comparing -1.75 with -1.6, if you use the representation -2+0.25, -2+0.4,  0.4 > 0.25 gives you the correct comparison. 

If they were represented as -1-0.75 and -1-0.6 then -0.6 > 0.75 would still give you the correct comparison.

I'll take out the panic.